src/rdkafka_topic.c

```diff
@@ -629,12 +629,15 @@ static int rd_kafka_toppar_leader_update (rd_kafka_topic_t *rkt,
                         leader_id, rktp->rktp_broker_id);
                 r = 0;
         } else {
-                rktp->rktp_leader_id = leader_id;
-                if (rktp->rktp_leader)
-                        rd_kafka_broker_destroy(rktp->rktp_leader);
-                if (leader)
-                        rd_kafka_broker_keep(leader);
-                rktp->rktp_leader = leader;
+                if (rktp->rktp_leader_id != leader_id) {
+                        rktp->rktp_leader_id = leader_id;
+                        if (rktp->rktp_leader)
+                                rd_kafka_broker_destroy(rktp->rktp_leader);
+                        if (leader)
+                                rd_kafka_broker_keep(leader);
+                        rktp->rktp_leader = leader;
+                }
+
```

Comment:
the if statement above is that case AND fetching from a replica. they could be equal here if not fetching from a replica. have made the logic clearer.

Reply:
yes you're right, have changed it. it's not incorrect as i have it because the update method does nothing in the case there is no change. note: I arrived at this code by modifying existing logic in way that would remain correct assuming it already was.